Submitted By:            Pierre Labastie <pierre dot labastie at neuf dot fr>
Date:                    2026-06-12
Initial Package Version: 26.2.4.2
Origin:                  Marty J.
Upstream Status:         Applied
Description:             Fixes building Libreoffice with poppler-26.06
diff -Naur libreoffice-26.2.4.2.orig/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx libreoffice-26.2.4.2/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
--- libreoffice-26.2.4.2.orig/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx	2026-06-01 16:24:55.000000000 -0400
+++ libreoffice-26.2.4.2/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx	2026-06-05 08:15:53.742947929 -0400
@@ -1224,7 +1224,7 @@
             GfxRGB aMinRGB;
             colorMap->getColorSpace()->getRGB(
 #if POPPLER_CHECK_VERSION(0, 82, 0)
-                reinterpret_cast<const GfxColor*>(maskColors),
+                *(reinterpret_cast<const GfxColor*>(maskColors)),
 #else
                 reinterpret_cast<GfxColor*>(maskColors),
 #endif
@@ -1233,7 +1233,7 @@
             GfxRGB aMaxRGB;
             colorMap->getColorSpace()->getRGB(
 #if POPPLER_CHECK_VERSION(0, 82, 0)
-                reinterpret_cast<const GfxColor*>(maskColors)+gfxColorMaxComps,
+                *(reinterpret_cast<const GfxColor*>(maskColors)+gfxColorMaxComps),
 #else
                 reinterpret_cast<GfxColor*>(maskColors)+gfxColorMaxComps,
 #endif
@@ -1365,7 +1365,7 @@
     aBox.y2 = pBbox[3];
 
     const int nDPI = 72; // GfxState seems to have 72.0 as magic for some reason
-    auto pSplashGfxState = new GfxState(nDPI, nDPI, &aBox, 0, false);
+    auto pSplashGfxState = new GfxState(nDPI, nDPI, aBox, 0, false);
 #if POPPLER_CHECK_VERSION(26, 2, 0)
     auto pSplashOut = new SplashOutputDev(splashModeRGB8, 1, nullptr);
 #else
@@ -1375,13 +1375,11 @@
     pSplashOut->startDoc(m_pDoc);
     pSplashOut->startPage(0 /* pageNum */, pSplashGfxState, nullptr /* xref */);
 
-    auto pSplashGfx = new Gfx(m_pDoc, pSplashOut, pResDict, &aBox, nullptr);
+    auto pSplashGfx = new Gfx(m_pDoc, pSplashOut, pResDict, aBox, nullptr);
     pSplashGfx->display(aStr);
     std::unique_ptr<SplashBitmap> pSplashBitmap(pSplashOut->takeBitmap());
     // Poppler tells us to free the splash device immediately after taking the
     // bitmap
-    delete pSplashGfxState;
-    delete pSplashGfx;
     delete pSplashOut;
 
     // Add a vertical flip, we can't do this in LO for an image filled poly
diff -Naur libreoffice-26.2.4.2.orig/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx libreoffice-26.2.4.2/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx
--- libreoffice-26.2.4.2.orig/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx	2026-06-01 16:24:55.000000000 -0400
+++ libreoffice-26.2.4.2/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.hxx	2026-06-05 08:15:53.749046066 -0400
@@ -37,6 +37,7 @@
 #include <GfxFont.h>
 #include <UnicodeMap.h>
 #include <Link.h>
+#include <Annot.h>
 #include <Object.h>
 #include <OutputDev.h>
 #include <GlobalParams.h>
